### DISCUSSION TITLE
Can I trust VSG today in the implementation of a commercial project

### DISCUSSION THREAD

**Discussion by maisvendoo** [vsg-dev/VulkanSceneGraph#1080] [Q&A] [answered]
I used OSG by two projects - first is commercial, second - open sourse project. Today I want move on - I'll want use VSG in new project. However, I doubt the stability of the project for long-term development. How I know, today vesion 1.0.9 is stable version 1.1.0 is not stable. Can I hope than version 1.0.9 will capable with future version 1.2.0 next stable?

P.S: The VulkanSceneGraph project is relatively young and little known in our country. However, unlike the previous project, OpenSceneGraph, the developers took the trouble to create detailed online documentation. As a developer interested in the development of VSG, I had a desire to create a resource that would contain a translation of the official documentation into Russian.

The content of this site is mainly a free translation of the official documentation located on the site www.vulkanscenegraph.org into Russian. Perhaps, as I understand the capabilities of VulkanSceneGraph, my author’s publications will appear on this resource, supplementing the official documentation. [http//vulkanscenegaph](http://vulkanscenegraph.ru/)

**Comment by robertosfield** [comment, marked as answer]
The VSG has been used in commercial applications well before the 1.0.0 release in November 2022.  

While VSG-1.0.9 is the last "stable" release, it doesn't meant that VSG master or the VSG1.1.x development series won't be unreliable in comparison.

As porting from the OSG to VSG will likely be a reasonable block of work I would recommend just using VSG master or the 1.1.x series as a base as this will be cloest to the feature set in 1.2.  While the VSG doesn't attempt to be directly compatible with the OSG as time goes on and we expand the feature set it becomes easier for a wider range of application developers to port across OSG.

For instance if you need shadows then 1.1.0 or VSG master will be your go to.

If you want animation support then what I'm working on right now will be helpful.

There are also 3rd party projects that are building on-top of the VSG that are expanding the feature set.

What features from the OSG do you use right now?

**Comment by maisvendoo** [comment]
Thank you, your answer really encouraged me! I really want to promote your project in our country and at my university. I’m placing a serious bet on it, since this is an advanced development that few people know about yet.

Regarding your question about the list of OSG functions that I use. Here is a list of OSG libraries with which I link

LIBS += -L$$OSG_LIB_DIRECTORY -losg
LIBS += -L$$OSG_LIB_DIRECTORY -losgViewer
LIBS += -L$$OSG_LIB_DIRECTORY -losgDB
LIBS += -L$$OSG_LIB_DIRECTORY -lOpenThreads
LIBS += -L$$OSG_LIB_DIRECTORY -losgGA
LIBS += -L$$OSG_LIB_DIRECTORY -losgUtil
LIBS += -L$$OSG_LIB_DIRECTORY -losgText

accordingly, using their capabilities in the project

**Reply by robertosfield** [reply]
This is a reasonably narrow set so even VSG-1.0.0 will likely provide most of the functionality you need.  While not all the functionality found in each of these core OSG libs is provided by the VSG, even if the functionality isn't directly supported there will likely be ways of achieving the same end result.  Rather than need to link to all these libraries you'll just need to link to libvsg.

With modern AI coming along I wonder if translators will be able to translate documentation automatically without needing to have a human doing the translation and then keep it up to date.  The VSG documentation will be evolving over the next year or so as we build out all the most common features and then document it so having an automated mechanism would help avoid stuff getting out of date.
